### ISSUE TITLE
Correct `features` syntax in frontmatter

### ISSUE THREAD

**Issue by mathiasbynens** [tc39/test262#1817]
I was doing a Test262 roll for V8 and had lots of new failing tests because of this issue.

https://github.com/tc39/test262/blob/7e65999731cb6cd6fab10a92e8b7bd39a6b0793c/test/language/expressions/class/dstr-private-gen-meth-dflt-ary-init-iter-no-close.js#L7-L8

This `features` value is spread across multiple lines, which https://github.com/tc39/test262/blame/master/tools/packaging/parseTestRecord.py doesn’t seem to handle very well. It seems to ignore the `features` altogether in this case.

If you need to spread it across multiple lines, keep in mind that YAML offers this syntax:

```
features: 
  - Symbol.iterator
  - class
  - class-methods-private
  - generators
  - destructuring-binding
  - default-parameters
```

There are three options here:

1. stick to the `features: [foo, bar]` syntax on a single line
1. update `tools/packaging/parseTestRecord.py` to handle the `features: [foo, bar]` syntax spread across multiple lines
1. use the multiline syntax (and verify that `tools/packaging/parseTestRecord.py` supports it)

Ref. https://github.com/tc39/test262/pull/1708 and others.

**Comment by mathiasbynens** [comment]
These tests are affected:

```
test/language/statements/class/dstr-private-gen-meth-dflt-ary-init-iter-close.js test/language/statements/class/dstr-private-gen-meth-dflt-ary-init-iter-no-close.js test/language/statements/class/dstr-private-gen-meth-dflt-ary-ptrn-rest-id-exhausted.js test/language/statements/class/dstr-private-gen-meth-static-dflt-ary-init-iter-close.js test/language/statements/class/dstr-private-gen-meth-static-dflt-ary-init-iter-no-close.js test/language/statements/class/dstr-private-gen-meth-static-dflt-ary-ptrn-rest-id.js test/language/statements/class/dstr-private-gen-meth-static-dflt-obj-ptrn-rest-getter.js test/language/statements/class/dstr-private-gen-meth-static-dflt-obj-ptrn-rest-skip-nonmerable.js test/language/statements/class/dstr-private-gen-meth-static-dflt-obj-ptrn-rest-val-obj.js
```

**Comment by leobalter** [comment]
> If you need to spread it across multiple lines, keep in mind that YAML offers this syntax (...)

We are pretty familiar with this syntax but also the fact that some different runners are not fully compatible with the complete Yaml syntax and this might cause trouble I prefer to avoid for now.

It sounds like item 2 is the best candidate, IMO. It's also unfortunate this error is happening. Thanks for showing it.

> These tests are affected:

Mind breaking them in multiple lines? :)

**Comment by rwaldron** [comment]
@mathiasbynens who/what is using the file you mention here?

> This features value is spread across multiple lines, which https://github.com/tc39/test262/blame/master/tools/packaging/parseTestRecord.py doesn’t seem to handle very well. It seems to ignore the features altogether in this case.

**Comment by mathiasbynens** [comment]
V8’s test runner. See https://chromium-review.googlesource.com/c/v8/v8/+/1259865.

**Comment by rwaldron** [comment]
@mathiasbynens when following that link, I don't see where https://github.com/tc39/test262/blame/master/tools/packaging/parseTestRecord.py is being used

**Comment by mathiasbynens** [comment]
Right, the CL shows the failing tests because of this. It’s used in `testcfg.py`: https://cs.chromium.org/chromium/src/v8/test/test262/testcfg.py?l=117&rcl=573561bf18181e6a08b8db00576dcb8adc969800

Without that CL, you’ll see those tests failing when running `gm x64.release test262` in a V8 checkout.

Note that our [`SKIPPED_FEATURES`](https://cs.chromium.org/chromium/src/v8/test/test262/testcfg.py?q=file:%5Esrc/v8/+parsetestrecord&g=0&l=59) contains `class-methods-private`, so it should skip tests using that `feature`. However, due to the multi-line thingy, [`case.test_record.get("features", [])` results in `[]` instead of the list of features](https://cs.chromium.org/chromium/src/v8/test/test262/testcfg.py?q=file:%5Esrc/v8/+parsetestrecord&g=0&l=147-149). I’ve confirmed that changing the tests to fit the `features` on a single line fixes the problem.

**Comment by rwaldron** [comment]
I've determined that the culprit is: 

```js
        if len(features):
            lines += ['features: ' + yaml.dump(features).strip()]
```

Here: https://github.com/tc39/test262/blob/master/tools/generation/lib/template.py#L156-L157

And working on fixing

**Comment by mathiasbynens** [comment]
@rwaldron Cool! Sounds like that would option 1 then? (I’m fine with either of the three.)

**Comment by mathiasbynens** [comment]
Patch: https://github.com/tc39/test262/pull/1821
